### ISSUE TITLE
Inject nodes and Debug nodes are not executable, and nodes that query HA entity states are not displaying state under nodes.

### ISSUE THREAD

**Issue by twister36** [node-red/node-red#4902]
# Please see to the HA Add-on Issue: https://github.com/hassio-addons/addon-node-red/issues/1960

_Update from @knolleary:_ This is not a core Node-RED issue. We're keeping this issue open for HA users who come to us looking for help.

### Current Behavior

Inject and debug nodes appear to be disabled when you open the Node-Red web UI through Home Assistant (Setttings/Add-ons/Node-Red, then click "Open web UI"). Additionally, nodes that acquire state, such as the event.state, poll-state, etc no longer show the state below the node(s) (See FIG-1) 

FIG-1
![image](https://github.com/user-attachments/assets/ad6e7446-65d3-4602-a9e6-73cf8c2049a6)

However, if you open node-red web ui directly, through port 1800, inject, debug and nodes that interact with Home Assistant nodes are working as expected (See FIG-2)

FIG-2
![image](https://github.com/user-attachments/assets/543f19ef-371c-48c7-9764-2b1b06230eb6)

### Expected Behavior

Inject and debug nodes should work through the embeded HA web ui. The nodes that interact with Home Assistant entities should show the state. 

### Steps To Reproduce

Open, HA, Go to Node-Red add-on, open Node-Red Web UI. 

Issue 1: Once loaded, all inject nodes, debug nodes and nodes are disabled. Unable to execute these nodes.

Issue: 2: Nodes that use home assistant entities are not displaying state of the entity below the node. Flows still appear to be working. This appears to be a visual problem.  

### Example flow

All Flows are experiencing this issue. 

### Environment

- Node-RED version: 4.0.3
- Node.js version: where do I get this?
- npm version: where do I get this?
- Platform/OS: Home Assistant 2024.10.0
- Browser: Chrome Version 128.0.6613.138 (Official Build) (64-bit)

**Comment by bill7531** [comment]
I have the same issue using the same versions. Existing flows still work. Only some sensor entities are showing up when trying to search inside a node. Manually typing in entity name does not work. For example, using event:state now, manually typing in entity name and changing state does trigger flow.

**Comment by twister36** [comment]
> I have the same issue using the same versions. Existing flows still work. Only some sensor entities are showing up when trying to search inside a node. Manually typing in entity name does not work. For example, using event:state now, manually typing in entity name and changing state does trigger flow.

Does your node-red work when you directly access it via port 1800?

**Comment by bill7531** [comment]
> > I have the same issue using the same versions. Existing flows still work. Only some sensor entities are showing up when trying to search inside a node. Manually typing in entity name does not work. For example, using event:state now, manually typing in entity name and changing state does trigger flow.
> 
> Does your node-red work when you directly access it via port 1800?

yes, it seems to work when accessed that way. I am restoring to 4.0.2 until this is sorted.

**Comment by hardillb** [comment]
You should raise this with the Home Assistant team

**Comment by bill7531** [comment]
Looks like it’s been raised

> You should raise this with the Home Assistant team, this is most likely a reverse proxying problem as the buttons and status all depend on the websocket comms link.
> 
> Just to be clear, the core Node-RED team have no connection to Home Assistant or the group that package it for Home Assistant

Looks like it has been raised 

https://github.com/hassio-addons/addon-node-red/issues/1960

**Comment by knolleary** [comment]
Thanks for the link to the HA issue.

I'll keep this issue open for the time being for other HA users to find if they come searching,

**Comment by hardillb** [comment]
All I will add is that it may be a CORS problem as the HA issue mentions an iframe

**Comment by dvine2000** [comment]
Saw this issue myself and on HA forum talking.

My Solution >>>>>>>>>>>>>>>

Log into nod red with local IP of HA and port number
http://ha-ip:1880/

I saw some errors.
All of them where nodes or entity’s of old devices or not connected nodes.
I resolve them all and after this I could use the inject and debug again.

![image](https://github.com/user-attachments/assets/02031f2e-1b07-4ec4-b162-68171fca43c8)

**Comment by redlegoman** [comment]
just not the case for me, before I update and after I update, all seems fine except the inject nodes can no longer be pressed

![image](https://github.com/user-attachments/assets/4d5d97e5-4d4a-4033-b1fb-c4ce2e105d48)

![image](https://github.com/user-attachments/assets/6ed72fdf-43eb-406f-9273-820e28b67aa8)

**Comment by sanderlv** [comment]
Issue discussed here too: https://github.com/zachowj/node-red-contrib-home-assistant-websocket/issues/1480#issuecomment-2403337429

**Comment by helmut-hackl** [comment]
Solution is found here:

https://github.com/zachowj/node-red-contrib-home-assistant-websocket/issues/1480#issuecomment-2434005277

**Comment by alexruffell** [comment]
I too had the issue where the debug and inject nodes were not working. The latest release of the WebSocket (I believe this is what fixed it) resolved the problem for me. However, I am still experiencing issues that I believe are related, as the symptoms are similar. For example, I wasn’t getting any output in the debug pane, so I had two Node-RED windows open - one using port 1880 and the other through Home Assistant. On the 1880 tab, I could see the debug output, but the action node was missing the target fields. On the Home Assistant tab, I couldn’t see the debug output, but the target fields were present inside the action node.

I have don't lots of searching but can't find any viable solution and this is getting pretty frustrating.

**Comment by alexruffell** [comment]
@dceejay - I installed NR years ago using the HA add-on method so as far as I know all is "default". On a HA system you can access NR both via port 1880 that accesses NR directly (no HA shown on screen), or via HA which runs on port 8123 and loads NR in an iframe. Anyhow the issues I am having are same or similar to those in this thread although they have evolved a bit as I tried to explain in my previous post. I realize this is likely not a NR issue specifically but something related to the NR integration within HA.... I was just adding my latest observations on the issue in case it might help. Per your suggestion, I will hunt for configuration files in the NR folder to see if anything seems off.

**Comment by zen2** [comment]
I do not have this problem but I'm using a stand alone node-red installation that is integrate in HA as a dashboard.
So you can easily access your node-red within HA by simply adding a web page dashboard using your node-red URL as workaround.

**Comment by AlexKalopsia** [comment]
Experiencing the same problem. 

If I visit `host.mydomain.com:1880`, I can access NodeRed and interact with the buttons
If I visit `nodered.host,mydomain.com` I can access NodeRed but the buttons can't be clicked

**Comment by knolleary** [comment]
Closing as part of a tidy-up of the issue tracker — this has been inactive for a long time and much has changed in Node-RED since. If it's still relevant on a current version, please open a fresh issue with up-to-date details and we'll take a look. Thanks for raising it.
